lib/config/sfFactoryConfigHandler.class.php

```diff
@@ -109,6 +109,12 @@ public function execute($configFiles)
             unset($parameters['database']);
           }
 
+          $sessionGcMaxlifetime = 1800;
+          if (isset($config['user']['timeout'])) {
+            $sessionGcMaxlifetime = $config['user']['timeout'];
+          }
+          $defaultParameters[] = sprintf("'gc_maxlifetime' => %d,", $sessionGcMaxlifetime);
```

Comment:
Only append this new configuration if it is set on user factory configuration.

Let keep the default configuration value on `sfSessionStorage` class to avoid duplication and confusion.